File: node/packages/glaive/lib/input_string.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InputString = void 0;

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

var InputString =
/*#__PURE__*/
function () {
  function InputString() {
    _classCallCheck(this, InputString);

    this.pos = 0;
  }

  _createClass(InputString, [{
    key: "read",
    value: function read() {
      var me = this;
      var char = str[me.pos++];
      if (!char) return -1;else return char.charCodeAt(0);
    }
  }]);

  return InputString;
}();

exports.InputString = InputString;
;

File: go/src/github.com/w3esoft/input/inputstring/inputstring.go
package inputstring


type InputString struct {
	value string
	pos   int
}
func  New(value string) *InputString {
	i := InputString{}
	i.pos = 0
	i.value=value;
	return &i
}
func (input *InputString ) Read() byte{
	l :=len(input.value)
	if l > input.pos {
		value  :=  input.value[input.pos]
		input.pos =input.pos+1
		return value;
	} else{
		return 0;
	}
}
func (input *InputString ) GetPosition() int{
	l :=len(input.value)
	if l > input.pos{
		return input.pos;
	} else{
		return l;
	}
}
